python/tk_framework_desktopserver/server_protocol.py

```diff
@@ -157,6 +142,34 @@ def onMessage(self, payload, isBinary):
 
         self._protocol_version = message["protocol_version"]
 
+        if self._protocol_version == 2:
+
+            # Version 2 of the protocol can't only answer requests from the site and user the server
+            # is authenticated into. Validate this.
+
+            # origin is formatted such as https://xyz.shotgunstudio.com:port_number
+            # host is https://xyz.shotgunstudio.com:port_number
+            origin_network = self._origin.lower()
+            host_network = self.factory.host.lower()
+
+            # Try to get the user information. If that fails, we need to report the error.
+            try:
+                user_id = message["command"]["data"]["user"]["entity"]["id"]
+            except Exception:
+                logger.exception("Unexpected error while trying to retrieve the user id.")
+                self.sendClose(3000, u"No user information was found in this request.")
```

Comment:
Twisted documentation says application code errors start at 3000. Let's do this.